src/lib/uTop_compat.ml

```diff
@@ -19,11 +19,26 @@ let toploop_all_directive_names () =
   Hashtbl.fold (fun dir _ acc -> dir::acc) Toploop.directive_table []
 #endif
 
-let set_load_path path =
-#if OCAML_VERSION >= (5, 0, 0)
-  Load_path.init path ~auto_include:Load_path.no_auto_include
+type paths = { visible : string list; hidden : string list; }
+
+let get_load_path () =
+#if OCAML_VERSION >= (5, 2, 0)
+  let paths = Load_path.get_paths () in
+  { visible = paths.visible; hidden = paths.hidden }
+#elif OCAML_VERSION >= (4, 14, 0)
+  { visible = Load_path.get_paths (); hidden = [] }
+#else
+  { visible = Load_path.get_load_path (); hidden = [] }
+#endif
+
+let set_load_path paths =
+#if OCAML_VERSION >= (5, 2, 0)
+  let { visible; hidden } = paths in
+  Load_path.init ~auto_include:Load_path.no_auto_include ~visible ~hidden
+#elif OCAML_VERSION >= (5, 0, 0)
+  Load_path.init ~auto_include:Load_path.no_auto_include paths.visible
 #else
-  Load_path.init path
+  Load_path.init paths.visible
```

Comment:
@emillon Would this be`visible @@ hidden`

Reply:
Does utop need a `~hidden` flag like the compiler? Ref: https://github.com/ocaml/ocaml/commit/01178bb60eb337790c597af937ac79419eab241c